envi/archs/arm/disasm.py

```diff
@@ -597,10 +600,15 @@ def p_dp_reg_shift(opval, va):
 def p_mult(opval, va):
     ocode, vals = chopmul(opval)                         
     mnem, opindexes, flags = iencmul_codes.get(ocode, multfail)
+    #work around because masks match up - should be a cleaner way to do this?
+    #if Ra = 15 then smmul
     if mnem == None:
         raise envi.InvalidInstruction(
                 mesg="p_mult: invalid instruction",
                 bytez=struct.pack("<I", opval), va=va)
+    elif vals[1] == 15 and(mnem == 'smmla' or mnem == 'smmlar'):
```

Comment:
Thought you might have an issue with it but wasn't sure.. figured it would be faster the way I did it than the other way around. Your way is more elegant.